openshift/ci-operator/build-image/setup-logging-for-e2e.sh

```diff
@@ -6,28 +6,164 @@
 # https://github.com/openshift/release/blob/master/ci-operator/templates/cluster-launch-installer-src.yaml
 # where the script will sourced
 
-# it is expected the image was built with golang
-# but all of this operator stuff could go away if the ci test is changed to
-# deploy logging using OLM and a subscription
-gopath=$( go env GOPATH | cut -f1 -d: )
-if [ ! -d $gopath/src/github.com/openshift/elasticsearch-operator ] ; then
-    git clone https://github.com/${ESO_OPERATOR_REPO:-openshift}/elasticsearch-operator \
-    --branch ${ESO_OPERATOR_BRANCH:-master} \
-    $gopath/src/github.com/openshift/elasticsearch-operator
-fi
-if [ ! -d $gopath/src/github.com/openshift/cluster-logging-operator ] ; then
-    git clone https://github.com/${CLO_OPERATOR_REPO:-openshift}/cluster-logging-operator \
-    --branch ${CLO_OPERATOR_BRANCH:-master} \
-    $gopath/src/github.com/openshift/cluster-logging-operator
-fi
-testroot=$( pwd )
-pushd $gopath/src/github.com/openshift/cluster-logging-operator > /dev/null
-if [ -z "${ARTIFACT_DIR:-}" ] ; then
-    ARTIFACT_DIR=/tmp/artifacts
-    if [ ! -d $ARTIFACT_DIR ] ; then
-        mkdir -p $ARTIFACT_DIR
+logging_err_exit() {
+    oc get deploy >> ${ARTIFACT_DIR}/logging_err_exit.log 2>&1 || :
+    oc get pods >> ${ARTIFACT_DIR}/logging_err_exit.log 2>&1 || :
+    oc -n openshift-operators get elasticsearch >> ${ARTIFACT_DIR}/logging_err_exit.log 2>&1 || :
+    oc get clusterlogging >> ${ARTIFACT_DIR}/logging_err_exit.log 2>&1 || :
+    oc get crds >> ${ARTIFACT_DIR}/logging_err_exit.log 2>&1 || :
+    oc describe pods >> ${ARTIFACT_DIR}/logging_err_exit.log 2>&1 || :
+    for p in $( oc get pods -o jsonpath='{.items[*].metadata.name}' ) ; do
+        for container in $( oc get po $p -o jsonpath='{.spec.containers[*].name}' ) ; do
+            echo pod $p container $container >> ${ARTIFACT_DIR}/logging_err_exit.log 2>&1 || :
+            oc logs -c $container $p >> ${ARTIFACT_DIR}/logging_err_exit.log 2>&1 || :
+        done
+    done
+    oc get events >> ${ARTIFACT_DIR}/logging_err_exit.log 2>&1 || :
+    cat ${ARTIFACT_DIR}/test_output
+    exit 1
+}
+
+wait_for_condition()
+{
+    # $1 is shell function condition to execute until it returns success
+    # $2 is the timeout number of retries - default 60
+    # $3 is the interval in seconds - default 1
+    # e.g. the total timeout in seconds is timeout * interval
+    local cmd=$1
+    local timeout=${2:-60}
+    local interval=${3:-1}
+    local ii=0
+    for ii in $( seq 1 $timeout ) ; do
+        if $cmd ; then
+            break
+        fi
+        sleep $interval
+    done
+    if [ $ii = $timeout ] ; then
+        return 1
     fi
+    return 0
+}
+
+# Create the openshift-logging namespace:
+cat > openshift-logging-namespace.yaml << EOF
```

Comment:
Wouldn't it make more sense to define these in files and apply them directly instead of using the bash script?  I shortly will have to introduce a PR to redefine our `manifest` so we get into the Operator Marketplace.  Seems like there is an opportunity to re-use static files.